File: 2 semestre/Técnica de Desenvolvimento de Algoritmos/Aula 13 10-11-16/vetorPrimos.c
#include <stdio.h>

main()
{
	printf("Vetor de Numeros Primos\n\n");
	
	//vars
	int v[7], i, j, cont=0, contPrimo;
	
	//Entrada
	for( i=0; i<7; i++)
	{
		printf("Digite v[%d]: ", i);
		scanf("%d", &v[i]);
	}
	
	//Processamento
	for( i=0; i<7; i++ )
	{
		contPrimo = 0;
		for( j=1; j<=v[i]; j++ )
		{
			if( v[i]%j == 0 )
			{
				contPrimo++;
			}
		}
		
		//Se for primo
		if( contPrimo == 2 )
			cont++;
	}
	//Saída
	//Se houver primos
	if( cont > 0)
		for( i=0; i<7; i++)
		{
			contPrimo=0;
			for( j=1; j<=v[i]; j++)
			{
				if( v[i]%j == 0 )
				{
					contPrimo++;
				}
			}
			
			//Se v[i] for primo
			if( contPrimo == 2)
				printf("\n%d eh primo", v[i]);
			else
				printf("\n%d NAO eh primo", v[i]);
		}
	else
		printf("\n Nao ha primos!");
	
	
}

File: 2 semestre/Laboratório de Desenvolvimento de Algoritmos/UNICID - Semestre 2/src/aula11_08_11_16/VetorPrimos.java
//Importando Scanner
import java.util.Scanner;

public class VetorPrimos {

	public static void main(String[] args){
		
		//Instanciando Scanner
		Scanner input = new Scanner(System.in);
		
		//vars
		int v[] = new int[8];
		int vPrimo[] = new int[8];
		int i, j, k=0, cont;
		
		//Entrada
		for( i=0; i<8; i++ )
		{
			System.out.printf("Digite v[%d]: ", i);
			v[i] = input.nextInt();
		}
		
		//Processamento
		for( i=0; i<8; i++ )
		{
			//Calculando se v[i] é primo
			cont = 0;
			for( j=1; j<=v[i]; j++ )
			{
				if( v[i]%j == 0 )
					cont++;
			}
			
			//Atribuindo valores primos ao vPrimo
			if( cont == 2 )
			{
				vPrimo[k] = v[i];
				//k é contador do vPrimo
				k++;
			}
		}
		
		//Saída
		System.out.println("\nVetor dos números primos: ");
		if( k > 0)
		{
			for( i=0; vPrimo[i]>0; i++ )
			{
				System.out.printf("%d  ", vPrimo[i]);
			}	
		}
		else
		System.out.println("\nNão há primos!");
		
		
	}
}
